### ISSUE TITLE
filterTaps disables all click events in Jest

### ISSUE THREAD

**Issue by jantimon** [pmndrs/use-gesture#418]
**Describe the bug**

the `filterTaps` option works great in all browser however in JEST it prevents child listeners from working.

I am only guessing but maybe the following line doesn't work well as JEST is not able to get real size measurements:
https://github.com/pmndrs/use-gesture/blob/9217f29802b8e4d87dab42e2cbd7313d2179ffea/packages/core/src/engines/DragEngine.ts#L210

**Sandbox or Video**

I prepared this [CodeSandbox](https://codesandbox.io/s/cool-rhodes-27dr2?file=/src/Demo2.tsx:177-187) with unit tests.

It contains Demo1 and Demo2:

Demo1
```tsx
import { useDrag } from '@use-gesture/react'

export const Demo1 = ({onClick}: {onClick: () => void}) => {
  const bind = useDrag(({ down, movement: [mx, my] }) => {
  }, {

  })
  return <div {...bind()}>
      <button onClick={onClick}>
      click me
      </button>
    </div>
```

```tsx
import { useDrag } from '@use-gesture/react'

export const Demo2 = ({onClick}: {onClick: () => void}) => {
  const bind = useDrag(({ down, movement: [mx, my] }) => {
  }, {
    filterTaps: true
  })
  return <div {...bind()}>
      <button onClick={onClick}>
      click me
      </button>
    </div>
}
```

As you can see the only difference is `filterTaps: true`.

The result is that the unit test which tries to click the button fails for Demo2:

![testResult](https://user-images.githubusercontent.com/4113649/148423340-26e33387-46eb-4c45-aca1-57d0622c1a6f.png)

**Checklist:**
  - [x] I've read the [documentation](https://use-gesture.netlify.app/).

**Comment by dbismut** [comment]
Hey, thanks for reporting this.

The `filterTaps` option checks if the drag is an actual drag or a tap (it does this by checking whether the displacement is greater than `3px`). If the drag is a drag, then it prevents the `click` event to propagate (the `click` listener is added using `capture: true` so that events bubble down).

https://github.com/pmndrs/use-gesture/blob/9217f29802b8e4d87dab42e2cbd7313d2179ffea/packages/core/src/engines/DragEngine.ts#L231-L236

So I'm not sure how I would be able to solve this. Shouldn't this be a bug in Jest instead?

**Comment by jantimon** [comment]
After some further investigation I found out that `fireEvent.click` does not trigger `mouseDown` and `mouseUp` therefore `state.tap` is always false

There is a helper library which triggers a full click which works:

```ts
import userEvent from '@testing-library/user-event';

userEvent.click()
```

Could we set `state.taps = true` by default?

**Comment by dbismut** [comment]
Oh right. 

You mean setting `state.tap = false` on `pointerDown` then?

I'd have to make sure there's no collaterals.

**Comment by jantimon** [comment]
Rather here:

https://github.com/pmndrs/use-gesture/blob/9217f29802b8e4d87dab42e2cbd7313d2179ffea/packages/core/src/engines/DragEngine.ts#L29

**Comment by dbismut** [comment]
Yes, that's what I meant, but I would have to restore `state.tap` to a falsy value as soon as the gesture starts, as people might rely on it before `pointerUp`. In any case it wouldn't make sense to have `state.tap = true` when the user drags things around.

**Comment by jantimon** [comment]
Oh I see - good point
